sstables/storage.cc

```diff
@@ -114,6 +117,43 @@ future<> filesystem_storage::rename_new_file(const sstable& sst, sstring from_na
     });
 }
 
+future<> filesystem_storage::sync_cached_directory(std::string_view name) const {
+    std::string key(name);
+    auto cached_dir = _cached_dirs.find(key);
+
+    if (cached_dir == _cached_dirs.end()) {
+        auto dir_f = co_await open_directory(name);
+        std::exception_ptr ex;
+
+        try {
+            auto [inserted_at, has_been_inserted] = _cached_dirs.emplace(std::move(key), std::move(dir_f));
+
+            assert(has_been_inserted);
+            cached_dir = inserted_at;
+        } catch (...) {
+            ex = std::current_exception();
+        }
+
+        // A co_await expression cannot appear in an exception handler, so we
+        // need to handle the insertion failure here.
+        if (ex) {
+            sstlog.warn("Failed to insert {} into cache: {}", key, ex);
+            co_await dir_f.flush();
+            co_await dir_f.close();
+            co_return;
+        }
+    }
+
+    // Either found the directory in the cache, or inserted it successfully just
+    // now.
+    co_await cached_dir->second.flush();
+}
+
+future<> sync_cached_directory(const filesystem_storage* fsstor, std::string_view name)
+{
```

Comment:
style nit: `{` sticks to the function definition line

Reply:
fixed, thanks!

Reply:
no longer applies to v4, resolving